### ISSUE TITLE
Creating connection pool with "Reuse inbound user"

### ISSUE THREAD

**Issue by alexgo84** [aiven/aiven-go-client#334]
By following the requests from Aiven web console, I see that for "Reuse Inbound User" the request is sent with `null` value for the `username` request parameter.

However when I try to use the go client, the type `UpdateConnectionPoolRequest` has the `username` property as:
```go
	Username string `json:"username"`
```

When not specifying a username in the request (empty string), I receive the error:
```
Invalid input for username: Must consist of alpha-numeric characters, dots, underscores or dashes, may not start with dash or dot, max 64 characters
```

Since the `username` property is not a pointer and has no `omitempty` directive, it is not clear how to create a connection pool that isn't assigned to a specific user. Could I be missing something?

I imagine the same applies to the PUT endpoint.

**Comment by Serpentiel** [comment]
Hey, @alexgo84! 👋 

Thank you for raising this issue. I will be taking a look in order to supply a fix for it.

In the meantime, I would suggest checking if https://github.com/aiven/go-client-codegen works better for your needs.

We will be gradually moving to it, and this project will soon be deprecated, and replaced with the one that I linked.

If you haven't yet started using the client and are simply checking on the things, the new client should be the way to go. 🙂

**Comment by Serpentiel** [comment]
Hey again, @alexgo84! 👋

May I know what version of the Go client are you using? I would recommend upgrading to the latest version.

In the latest version, it's currently defined as:

```go
Username *string `json:"username"`
```

I can see that the fix was already introduced in #149.

I will close this issue for now. Should you have any additional information to share, or any questions or concerns, feel free to let me know by replying to this issue! 🙂
